### ISSUE TITLE
Feature request: adding proxy settings to Window and <webview> tag

### ISSUE THREAD

**Issue by Sunvas** [nwjs/nw.js#7955]
### Current Behavior
Entire NWjs application uses common proxy configuration https://nwjs.readthedocs.io/en/latest/References/App/#appsetproxyconfigconfig-pac_url.

### Expected/Proposed Behavior
Is it possible to specify proxy configuration for Window and/or <webview> tag.

**Comment by Sunvas** [comment]
@rogerwang any plans to implement this feature? ElectronJS already [has](https://www.electronjs.org/docs/latest/api/session#sessetproxyconfig) this feature.

**Comment by rogerwang** [comment]
@Sunvas after reading the link you post, I don't think Electron has this feature -- it sets the proxy for the whole session, not per window.

**Comment by Sunvas** [comment]
@rogerwang in Electron you can create session with proxy settings and assign it session to [BrowserWindow](https://www.electronjs.org/docs/latest/api/browser-window). There is such option in `webPreferences`.

**Comment by rogerwang** [comment]
Yeah a session can be assigned to a window, and each session can have its own proxy settings. However, it's important to clarify that this is fundamentally different from supporting per-window proxy configuration as a standalone feature.

The key distinction is that a session encompasses far more than just proxy settings—it includes things like cookies, localStorage, cache, and other persistent data. Assigning a different session to a window means you're isolating all of those aspects, not just the proxy. This can lead to unintended side effects for applications that want to share session state (like cookies or storage) across windows but route them through different proxies.

What the proposed feature aims for is more granular control: allowing each window to have its own proxy configuration without affecting or duplicating session-level state. This is not achievable through the session-based approach, which tightly couples proxy settings with broader browsing context.
